tests/acl/test_acl_outer_vlan.py

```diff
@@ -203,77 +240,53 @@ def setup_vlan(rand_selected_dut, vlan_setup_info):
-        cmds.append('config interface ip remove Vlan{} {}'.format(new_vlan, vlan_setup[new_vlan]['vlan_ip'][IPV6]))
-    rand_selected_dut.shell_cmds(cmds=cmds)
-    time.sleep(10)
-
-    # Remove testing vlan
-    cmds = []
-    for new_vlan in TEST_VLAN_LIST:
-        cmds.append('config vlan del {}'.format(new_vlan))
-    rand_selected_dut.shell_cmds(cmds=cmds)
-    time.sleep(10)
-
-    default_vlan_id = vlan_setup['default_vlan'].replace("Vlan", "")
-    # Add back interface to default Vlan
-    cmds = []
-    for port in list(test_ports.keys()):
-        cmds.append('config vlan member add {} {} --untagged'.format(default_vlan_id, port))
-    rand_selected_dut.shell_cmds(cmds=cmds)
-    time.sleep(10)
-
-
 @pytest.fixture(scope='module', autouse=True)
-def vlan_setup_teardown(rand_selected_dut, ptfadapter, vlan_setup_info, tbinfo):
+def vlan_setup_teardown(rand_selected_dut, vlan_setup_info, duthost, ptfhost):
     try:
-        setup_vlan(rand_selected_dut, vlan_setup_info)
+        setup_vlan(rand_selected_dut, vlan_setup_info, duthost, ptfhost)
         yield
     finally:
-        teardown_vlan(rand_selected_dut, vlan_setup_info)
+        _, _, _, portchannel_setup = vlan_setup_info
+        # Restore ptf configuration
+        ptfhost.set_dev_no_master(PTF_LAG_NAME)
+        for _, lag_data in portchannel_setup.items():
+            for idx in lag_data['ptf_member']:
+                ptf_lag_member = 'eth%u' % idx
+                ptfhost.set_dev_no_master(ptf_lag_member)
+                ptfhost.set_dev_up_or_down(ptf_lag_member, True)
+
+        ptfhost.shell("ip link del {}".format(PTF_LAG_NAME))
+        ptfhost.ptf_nn_agent()
+        # Wait for lag sync
+        time.sleep(10)
+        config_reload(duthost, safe_reload=True, check_intf_up_ports=True)
 
 
-def send_and_verify_traffic(ptfadapter, pkt, exp_pkt, src_port, dst_port, pkt_action=ACTION_FORWARD):
+def send_and_verify_traffic(ptfadapter, pkt, exp_pkt, src_port_list, dst_port_list, pkt_action=ACTION_FORWARD):
     """
     Send traffic and verify that traffic was received
 
     Args:
         ptfadapter: PTF adapter
         pkt: Packet that should be sent
         exp_pkt: Expected packet
-        src_port: Source port
-        dst_port: Destination port
+        src_port_list: Source port
+        dst_port_list: Destination port
         pkt_action: Packet action (forward or drop)
     """
     ptfadapter.reinit()
-    logger.info("Send packet from port {} to port {}".format(src_port, dst_port))
-    testutils.send(ptfadapter, src_port, pkt)
+    logger.info("Send packet from port {} to port {}".format(src_port_list, dst_port_list))
+    testutils.send(ptfadapter, src_port_list[0], pkt)
 
     if pkt_action == ACTION_FORWARD:
-        testutils.verify_packet(ptfadapter, exp_pkt, dst_port)
+        for dst_port in dst_port_list:
+            result = testutils.dp_poll(ptfadapter, port_number=dst_port, timeout=5, exp_pkt=exp_pkt)
```

Comment:
We may use `verify_packet_any_port` instead of looping the ports.

Reply:
updated.